x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/LookupFromIndexService.java

```diff
@@ -80,7 +80,8 @@ protected String getRequiredPrivilege() {
 
     private static void validateTypes(DataType inputDataType, MappedFieldType fieldType) {
         // TODO: consider supporting implicit type conversion as done in ENRICH for some types
-        if (fieldType.typeName().equals(inputDataType.typeName()) == false) {
+        String typeName = fieldType.typeName().equals("text") ? DataType.KEYWORD.typeName() : fieldType.typeName();
+        if (typeName.equals(inputDataType.noText().typeName()) == false) {
```

Comment:
I wonder if it is worth resolving `typeName` wirh `fromTypeName` here to avoid comparing strings?
Something like
```suggestion
        if (Objects.equals(DataType.fromTypeName(fieldType.typeName()).noText(), inputDataType.noText()) == false) {
```

Reply:
I'm actually thinking of deleting this method entirely, since it is shadowed by the newer validation happening in the Validator. In the current draft I fixed both methods, but in reality we only need the one.